### PR TITLE
Fix source validation for local URLs and SSH repos

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/7790

Tried `git@code.byted.org:ugc/OpenCV.git` from the issue and it worked.

/cc @Whirlwind

### CODE REVIEW

File: spec/unit/installer/pod_source_installer_spec.rb [hunk 1]
```diff
@@ -62,6 +62,30 @@ module Pod
...
+        @installer.install!
+        UI.warnings.length.should.equal(0)
+      end
+
+      it 'does not warn for SSH repositories on Github' do
+        @spec.source = { :git => 'git@github.com:kylef/test.git', :tag => '1.0' }
+        dummy_response = Pod::Downloader::Response.new
+        Downloader.stubs(:download).returns(dummy_response)
+        @installer.install!
+        UI.warnings.length.should.equal(0)
```
Comment (amorde) [inline spec/unit/installer/pod_source_installer_spec.rb]: This seems identical to the test above - is there expected to be different behavior if the URL host is `github.com`?

Comment (amorde) [reply spec/unit/installer/pod_source_installer_spec.rb]: How would `git@github.com:kylef/test.git` and `git@bitbucket.org:kylef/test.git` differ?

Comment (dnkoutso) [reply spec/unit/installer/pod_source_installer_spec.rb]: Well not so much actually...I used the same tests that were applied here https://github.com/CocoaPods/Core/pull/84/files

Comment (amorde) [reply spec/unit/installer/pod_source_installer_spec.rb]: Huh, that's interesting..

Comment (dnkoutso) [reply spec/unit/installer/pod_source_installer_spec.rb]: @amorde overall harmless I say :) but you were right to comment on it.

**Pull request merged**